crates/payload/primitives/src/lib.rs

```diff
@@ -0,0 +1,345 @@
+//! This crate defines abstractions to create and update payloads (blocks)
+
+#![doc(
+    html_logo_url = "https://raw.githubusercontent.com/paradigmxyz/reth/main/assets/reth-docs.png",
+    html_favicon_url = "https://avatars0.githubusercontent.com/u/97369466?s=256",
+    issue_tracker_base_url = "https://github.com/paradigmxyz/reth/issues/"
+)]
+#![cfg_attr(not(test), warn(unused_crate_dependencies))]
+#![cfg_attr(docsrs, feature(doc_cfg, doc_auto_cfg))]
+
+mod error;
+pub use error::{EngineObjectValidationError, PayloadBuilderError, VersionSpecificValidationError};
+
+/// Contains traits to abstract over payload attributes types and default implementations of the
+/// [`PayloadAttributes`] trait for ethereum mainnet and optimism types.
+mod traits;
+pub use traits::{BuiltPayload, PayloadAttributes, PayloadBuilderAttributes};
+
+mod payload;
+pub use payload::PayloadOrAttributes;
+
+use reth_primitives::ChainSpec;
+use serde::{de::DeserializeOwned, ser::Serialize};
+use std::fmt::Debug;
+
+/// Payload types
+pub trait PayloadTypes: DeserializeOwned + Serialize + Debug + Unpin + Send + Sync + Clone {
+    /// The RPC payload attributes type the CL node emits via the engine API.
+    type PayloadAttributes: PayloadAttributes + Unpin;
+
+    /// The payload attributes type that contains information about a running payload job.
+    type PayloadBuilderAttributes: PayloadBuilderAttributes<RpcPayloadAttributes = Self::PayloadAttributes>
+        + Clone
+        + Unpin;
+}
```

Comment:
can we remove this for now,

we can try to introduce this in a follow-up

Reply:
yes exactly, I'm stuck at this as well, since I don't have a good design to follow with.